packages/client/lib/sync/skeleton.ts

```diff
@@ -652,6 +656,10 @@ export class Skeleton extends MetaDBManager {
         numBlocksInserted = await this.chain.putBlocks([block], true)
       } catch (e) {
         this.config.logger.error(`fillCanonicalChain putBlock error=${(e as Error).message}`)
+        if (resetHead === true && oldHead !== null) {
```

Comment:
```suggestion
        if (oldHead !== null) {
```
i think this should be sufficient since old head is only assigned in reset scenarios